src/main/java/com/google/sps/servlets/Util.java

```
@@ -0,0 +1,145 @@
+    final String accessToken = getAccessToken(refreshToken, clientId, clientSecret);
+    ListDeviceResponse resp = getDevicesResponse(EMPTY_PAGE_TOKEN, accessToken);
+    final List<ChromeOSDevice> allDevices = new ArrayList<>(resp.getDevices());
+    while (resp.hasNextPageToken()) {
+      final String pageToken = (String) resp.getNextPageToken();
+      resp = getDevicesResponse(pageToken, accessToken);
+      allDevices.addAll(resp.getDevices());
+    }
+    return allDevices;
+  }
+
+  public static String getAPIKey() {
+    try {
+      File file = new File(Util.class.getResource(API_KEY_FILE).getFile());
+      String str = FileUtils.readFileToString(file);
+      return str;
+    } catch (IOException e) {
+      System.out.println(e.getMessage());
+    }
+    return EMPTY_API_KEY;
+  }
+
+  private static String getRefreshToken(String userId) throws IOException {
+    Query query = new Query("RefreshToken").setFilter(FilterOperator.EQUAL.of("userId", userId));
+    PreparedQuery results = datastore.prepare(query);
+    System.out.println(results.countEntities());
+    try {
+      Entity entity = results.asSingleEntity();
+      String refreshToken = (String) entity.getProperty("refreshToken");
+      return refreshToken;
+    } catch (PreparedQuery.TooManyResultsException e) {
+      System.out.println(e);
+      throw new IOException("Error while getting refresh token");
+    }
+  }
+
+  private static String getAccessToken(String refreshToken, String clientId, String clientSecret) throws IOException {
+    try {
+      GoogleTokenResponse response =
+      new GoogleRefreshTokenRequest(new NetHttpTransport(), new JacksonFactory(),
+        refreshToken, clientId, clientSecret).execute();
+      return response.getAccessToken();
+    } catch (TokenResponseException e) {
+      if (e.getDetails() != null) {
+        System.out.println(e.getMessage());
+      }
+    }
+    return INVALID_ACCESS_TOKEN;
```

It's better to just throw the exception, instead of returning an invalid token and allowing the exception to be thrown at a later stage that you may even not know where it is.